### PR TITLE
add multi client examples

### PR DESCRIPTION
**Pull request by edgarriba**
Add a couple of examples showing how to use multiple clients:
- [x] Simple multi client subscriber
- [x] Camera and GPS with a non-bullet proof sync to geo tag images

Goes with https://github.com/farm-ng/amiga-dev-kit/pull/200

### CODE REVIEW

File: py/examples/multi_client_subscriber/main.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            tasks.append(asyncio.create_task(self._subscribe(subscription)))
+        # wait for the subscribe routines to finish
+        await asyncio.gather(*tasks)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        prog="amiga-multi-client-subscriber", description="Example of subscribing to events."
+    )
+    parser.add_argument("--config", type=Path, required=True, help="The system config.")
```
Comment (Hackerman342) [inline py/examples/multi_client_subscriber/main.py:70]: We should have a common CLI arg here vs other examples (which use `--service-config`). I'm good with either cc @guilhermedemouraa

File: py/examples/multi_client_geoimage/config.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      "name": "oak0",
+      "port": 50010,
+      "host": "localhost"
+    },
+    {
+      "name": "multi_subscriber",
+      "subscriptions": [
+        {
+          "uri": {
+            "path": "/rgb",
```
Comment (Hackerman342) [inline py/examples/multi_client_geoimage/config.json:18]: Suggest to use a lower resolution image stream as the default (in general). Will reduce issues induced by running this on a remote PC
```suggestion
            "path": "/left",
```

File: py/examples/multi_client_geoimage/main.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                    event_image, image = await self.image_queue.get()
+                    stamp_image = get_stamp_by_semantics_and_clock_type(
+                        event_image, semantics="service/send", clock_type="monotonic"
+                    )
+                    if stamp_image is None:
+                        continue
+                    stamp_diff = abs(stamp_gps - stamp_image)
+
+                    # NOTE: define this threshold depending on the precision of your application
+                    if stamp_diff > 0.05:
```
Comment (Hackerman342) [inline py/examples/multi_client_geoimage/main.py:80]: I found I was almost never able to synchronize an image with such a low threshold when running remotely. I'm sure locally this value is fine. Suggest to bump it up or better to add as a CLI arg
```suggestion
                    if stamp_diff > 0.1:
```

**Review by Hackerman342 — APPROVED** [review]
Tested :ok_hand:

**Review by guilhermedemouraa — APPROVED** [review]
Tested and it worked just fine!
For the `client_geoimage` example, I was never able to get synchronized data. Perhaps it was because I was running it on my local PC. Recommend loosening the constraint for the `stamp_diff.`

On another note, for the `multi_client_subscriber` example, I think it would be nice to print the messages received instead of printing that a message was received (or maybe printing both).

File: py/examples/multi_client_subscriber/main.py (hunk 1 above)
Comment (edgarriba) [reply py/examples/multi_client_subscriber/main.py:70]: in this particular case is a "system" config not a particular service config

**Comment by edgarriba** [comment]
> On another note, for the multi_client_subscriber example, I think it would be nice to print the messages received instead of printing that a message was received (or maybe printing both).

the terminal gets too cluttered

**Pull request merged**